api/include/opentelemetry/nostd/list.h

```diff
@@ -153,92 +132,97 @@ class list
 
     std::memcpy(temp, data_, previous_size * sizeof(T));
 
-    /* Save old array pointer for later deallocation. */
-    T *old_data = data_;
-
+    // Deallocate memory used by previous array.
+    delete[] data_;
+    
     data_ = temp;
 
     data_[previous_size] = element;
 
-    /* Deallocate memory used by previous array. */
-    delete[] old_data;
   };
 
-  /* Insert an element at the front of the list. */
+  // Insert an element at the front of the list.
   void push_front(T element)
   {
     int previous_size = size_;
     size_++;
 
     T *temp = new T[size_];
 
-    /* Copy the original data, leaving space for one element
-     * at the front. */
+    // Copy the original data, leaving space for one element
+    // at the front.
     std::memcpy(temp + 1, data_, previous_size * sizeof(T));
 
-    /* Save old array pointer for later deallocation. */
-    T *old_data = data_;
+    delete[] data_;
 
     data_ = temp;
 
     data_[0] = element;
-
-    /* Deallocate memory used by previous array. */
-    delete[] old_data;
   };
 
-  /* Returns the last element and removes it from the list. */
+  // Returns the last element and removes it from the list.
   T pop_back()
   {
+
     T back_elem = data_[size_ - 1];
     size_--;
 
     T *temp = new T[size_];
 
-    /* Copy the original data.*/
+    // Copy the original data.
     std::memcpy(temp, data_, size_ * sizeof(T));
 
-    /* Save old array pointer for later deallocation. */
-    T *old_data = data_;
+    delete[] data_;
 
     data_ = temp;
 
-    /* Deallocate memory used by previous array. */
-    delete[] old_data;
-
     return back_elem;
   }
 
-  /* Returns the first element and removes it from the list. */
+  // Returns the first element and removes it from the list.
   T pop_front()
   {
     T front_elem = front();
     size_--;
 
     T *temp = new T[size_];
 
-    /* Copy the original data.*/
+    // Copy the original data.
     std::memcpy(temp, data_ + 1, size_ * sizeof(T));
 
-    /* Save old array pointer for later deallocation. */
-    T *old_data = data_;
+    delete[] data_;
 
     data_ = temp;
 
-    /* Deallocate memory used by previous array. */
-    delete[] old_data;
 
     return front_elem;
   }
 
-  /* Returns the number of elements in the list. */
+  // Returns the number of elements in the list. //
   int size() { return size_; }
 
   void operator=(const list &other_list)
   {
-    data_ = other_list.data_;
+    delete[] data_;
+    data_ = new T[other_list.size_];
+    memcpy(data_, other_list.data_, other_list.size_*sizeof(T));
     size_ = other_list.size_;
   }
+  
+  // Returns true if the lists are the same, false if they are not
+  bool operator==(const list &other_list)
+  {
+    if(size_ != other_list.size_){
+      return false;
+    }
+
+    if( memcmp(data_, other_list.data_, size_) == 0){
+      return true;
+    }
+    else{
+      return false;
+    }
```

Comment:
Generally prefer to return the result of a boolean statement directly over putting it in a conditional and then returning true / false on the corresponding branches.

```suggestion
    return (memcmp(data_, other_list.data_, size_) == 0);
```

Reply:
Changed to removed if statements.